### PR TITLE
Changed ScriptWidget QLineEdit to QComboDropbox that loads all .py fi…

### PR DESCRIPTION
**Pull request by zackorenberg**
Very simple feature. Instead of a QLineEdit for ScriptWidget (selects user script to run), I used a QComboBox that loads all *.py scripts in the selected file's directory, selectable via up/down arrows. This is useful for when a user is working with multiple script files.

### CODE REVIEW

**Review by zackorenberg — COMMENTED** [review]
I added support for the legacy version using QLineEdit

File: LabTools/CoreWidgets/ScriptWidget.py [hunk 1]
```diff
@@ -13,40 +12,62 @@
...
+import os
 
+LEGACY_SCRIPTWIDGET = False
 
 class ScriptWidget(QWidget):
-    """widget with a lineEdit and a browse button"""
+    """widget with a dropdown and a browse button"""
 
-    def __init__(self, parent=None):
+    def __init__(self, parent=None, LEGACY = None):
```
Comment (Bachibouzouk) [inline LabTools/CoreWidgets/ScriptWidget.py]:
```suggestion
    def __init__(self, parent=None, legacy=None):
```
function arguments should be small caps

File: LabTools/CoreWidgets/ScriptWidget.py [hunk 2]
```diff
@@ -13,40 +12,62 @@
...
         super(ScriptWidget, self).__init__(parent)
 
+        if LEGACY is None:
+            self.LEGACY = LEGACY_SCRIPTWIDGET
+        else:
+            self.LEGACY = LEGACY
```
Comment (Bachibouzouk) [inline LabTools/CoreWidgets/ScriptWidget.py]:
```suggestion
            self.LEGACY = legacy
```

Comment (Bachibouzouk) [reply LabTools/CoreWidgets/ScriptWidget.py]: @zackorenberg - you can "add suggestion to batch" with the lower right button under my suggestion to bind the suggestion together :)

File: LabTools/CoreWidgets/ScriptWidget.py [hunk 3]
```diff
@@ -82,14 +148,17 @@ def add_widget_into_main(parent):
 
     fname = IOTool.get_script_name(config_file_path=parent.config_file)
 
-    parent.widgets['ScriptWidget'].scriptFileLineEdit.setText(fname)
+    #parent.widgets['ScriptWidget'].scriptFileLineEdit.setText(fname)
```
Comment (Bachibouzouk) [inline LabTools/CoreWidgets/ScriptWidget.py]: please do not push code which is commented out :)
Possible solution for that now:
- checkout your branch: `git checkout feature/dropdown_script_selector`
- erase the line locally and stage the change with `git add LabTools/CoreWidgets/ScriptWidgets.py`
- melt this change in the previous commit: `git commit --amend`
- force push your changes on the remote (only do that when the other people working on the branch are aware of it, i.e. tell them beforehand) : `git push -f`

**Review by Bachibouzouk — CHANGES_REQUESTED** [review]
Thanks for improving LabGui @zackorenberg 

Comments on the code : 

If it is an improvement, then why would we need to keep a LEGACY version? I don't think this would be a breaking change with the previous version.  Having too many options to choose from makes things more difficult for developers and maintainers.
LineEdit was chosen in the past because we trusted our "standard user" to be geek enough to write down a path in an empty box. As long as the option to choose any path is still there, I see the combobox as an improvment of the comfort of the user without a decrease in usability (which I find good :))

Therefore I would be in favor to keep only the combobox :)

Meta comments on github workflow:

- Thanks for making a PR with not too many changes, it is really enjoyable to review :)
- You could have made two commits : 1 with legacy and 1 with only the new code (i.e. function only used by combobox)
- You can also modify the changelog in a commit, so the description of the changes are done now
- You could have opened an issue describing why you wanted to make these changes to the code and where we could have chatted about the possible solution, thus avoiding you to write the LEGACY code if it ends up not being used ;)

**Comment by zackorenberg** [comment]
@Bachibouzouk I have amended the changes you requested to ScriptWidget.py. I left the legacy version as an option because I felt there was no real reason to delete the old functionality of ScriptWidget.py, as the changes I made were ultimately small. Should anyone wish to use this module in the LabGUI library for whatever reason, they will have the option to use the LineEdit by invoking legacy=True in the constructor.

**Comment by zackorenberg** [comment]
@Bachibouzouk what do you think?

**Review by Bachibouzouk — APPROVED** [review]
I approve your changes :)

**Pull request merged**